app/src/androidTest/java/org/kiwix/kiwixmobile/main/TopLevelDestinationTest.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.main
+
+import androidx.test.rule.ActivityTestRule
+import org.junit.Rule
+import org.junit.Test
+
+class TopLevelDestinationTest {
+  @Rule
+  @JvmField
+  var activityTestRule = ActivityTestRule(KiwixMainActivity::class.java)
+
+  @Test
+  fun testTopLevelDestination() {
+    kiwixMain {
+      clickReaderOnBottomNav {
+      }
+      clickLibraryOnBottomNav {
+        clickFileTransferIcon {
+        }
+      }
+      clickDownloadOnBottomNav {
+        clickOnGlobeIcon {
+        }
+      }
+      inNavDrawer {
+        clickBookmarksOnNavDrawer {
+          clickOnTrashIcon()
+          assertDeleteBookmarksDialogDisplayed()
+        }
+      }
+      inNavDrawer {
```

Comment:
I think it might look better if inNavDrawer were private so only an internal implementation detail of the robot and the structure of the test is unchanged from how it was previously. Right now it looks like NavDrawer is a robot, and maybe we could make it so it is? I am just idly considering how it could work, theoretically a `private inner class` kind of expresses it well but it could just be overkill

Reply:
Well, I think that the working code should be inside robot and let the test do only what it says. Thoughts?

Reply:
I am not entirely clear on your proposal. my suggestion is
```
// in the test
clickBookmarksOnNavDrawer{
//stuff
}

//inside robo
fun clickBookmarksOnNavDrawer(etc){
inNavDrawer{
 clickBookmarks()
 bookmarks()
 pressBack()
}
```
 optionally making a private robot MainRobot.NavDrawer

Reply:
Yep, this'll look nice, I was misunderstanding with moving this function to test. 
But why do you want to make it a class, a private function would just suffice.

Reply:
If we make it a Robot we get the benefits of a Robot as the functions are scoped to NavRobot, you can only click on settings from that robot, like how you can only click the globe icon if you are first on the OnlineLibrary. We make it `private` so nobody else can use it aside from Main/TopLevel & `inner` says we must have an instance of Main/TopLevel for this class to exist(and we can access members of the outer class but that is irrelevant). Still quite overkill, not something we have to do

Reply:
Yep I totally get it. But I don't think it is going to be used at some other class/test other than the current one. Okay, what about the current scenario in the robot, it looks fine to me, isn't it?

Reply:
Yeah the current scenario is fine, just the other scenario is better because we have Typed the problem space, reducing potential misuse but also comes with a larger overhead in time/cognition. Just highlighting how we could have done it.

Reply:
I'll be happier in writing the better implementation disregarding the time it takes, this PR is open for a long time, if it gets delayed by some more time for something good, I have no problem and will make the changes needed.

But don't you think, by making it a robot you'll contradict soomething what you said [here](https://github.com/kiwix/kiwix-android/pull/2431#issuecomment-714322426)?

Reply:
I was trying to show the ideas and responsibilities of a robot to you while also trying to move away from the idea of nesting everything in the 1 class. In this instance a private nested inner class make sense for the nav drawer, less so the bottom nav.  If you want to make the side nav robot go on ahead

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/2431/commits/82a63f311c8ae2b9f8cb3d17429992d31a9ed8df)